cmd/serviceusers/create.go

```diff
@@ -0,0 +1,176 @@
+package serviceusers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/appgate/sdp-api-client-go/api/v17/openapi"
+	"github.com/appgate/sdpctl/pkg/docs"
+	"github.com/appgate/sdpctl/pkg/factory"
+	"github.com/appgate/sdpctl/pkg/filesystem"
+	"github.com/appgate/sdpctl/pkg/prompt"
+	"github.com/appgate/sdpctl/pkg/util"
+	"github.com/hashicorp/go-multierror"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func NewServiceUsersCreateCMD(f *factory.Factory) *cobra.Command {
+	opts := ServiceUsersOptions{
+		Config: f.Config,
+		API:    f.ServiceUsers,
+		In:     f.Stdin,
+		Out:    f.IOOutWriter,
+	}
+	cmd := &cobra.Command{
+		Use:     "create",
+		Short:   docs.ServiceUsersCreate.Short,
+		Long:    docs.ServiceUsersCreate.Long,
+		Example: docs.ServiceUsersCreate.ExampleString(),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.Config.Version <= 16 {
+				return fmt.Errorf("The service user interface is only available from API version 17 or higher. Currently using API version %d", opts.Config.Version)
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return serviceUserCreateRun(cmd, args, opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.String("name", "", "name for service user")
+	flags.StringSlice("tags", []string{}, "tags for service user")
+	flags.StringP("from-file", "f", "", "create a user from a valid json file")
+
+	return cmd
+}
+
+func serviceUserCreateRun(cmd *cobra.Command, args []string, opts ServiceUsersOptions) error {
+	ctx := context.Background()
+	api, err := opts.API(opts.Config)
+	if err != nil {
+		return err
+	}
+
+	fromFile, err := cmd.Flags().GetString("from-file")
+	if err != nil {
+		return err
+	}
+
+	users := []openapi.ServiceUsersGetRequest{}
+	if len(fromFile) > 0 {
+		path := filesystem.AbsolutePath(fromFile)
+		ok, err := util.FileExists(path)
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return fmt.Errorf("file not found: %s", path)
+		}
+		file, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+		// Try unmarshal json array first
+		dtoArray := ServiceUserArrayDTO{}
+		if err := json.Unmarshal(file, &dtoArray); err != nil {
+			logrus.WithError(err).Warn("failed to unmarshal json to array")
```

Comment:
this seems un-nessesary to log this, given this flow for example:

```bash
sdpctl service-users create --no-interactive --from-file users.json
WARN[0000] failed to unmarshal json to array             error="json: cannot unmarshal object into Go value of type serviceusers.ServiceUserArrayDTO"
New service user created:
{
  "created": "2022-09-28T06:25:43.256229Z",
  "disabled": false,
  "failedLoginAttempts": 0,
  "id": "e4818be0-eb89-4180-809e-01f1aafefb27",
  "labels": {},
  "name": "daniel",
  "notes": "",
  "tags": [],
  "updated": "2022-09-28T06:25:43.256229Z"
}
~                                                                                                                                                                                                           8:25
> echo $?                                       
0
~                                                                                                                                                                                                           8:25
> jq . users.json
{
  "name": "daniel",
  "password": "abc123"
}

```

Reply:
removed it